src/model/writer.js

```diff
@@ -63,12 +70,27 @@ export function insert( position, nodes ) {
 
 	// Insert nodes at given index. After splitting we have a proper index and insertion is between nodes,
 	// using basic `Element` API.
-	parent.insertChildren( index, nodes );
+	parent.insertChildren( index, normalizedNodes );
 
 	// Merge text nodes, if possible. Merging is needed only at points where inserted nodes "touch" "old" nodes.
-	_mergeNodesAtIndex( parent, index + nodes.length );
+	_mergeNodesAtIndex( parent, index + normalizedNodes.length );
 	_mergeNodesAtIndex( parent, index );
 
+	// If given element is a DocumentFragment and has markers.
+	if ( nodes instanceof DocumentFragment && nodes.markers.size ) {
+		// If node is being inserted to the Document with markers collection.
+		if ( position.root.document && position.root.document.markers instanceof MarkerCollection ) {
+			// We need to transfer its markers and update position markers positions.
+			for ( const marker of nodes.markers ) {
+				const range = new Range( new Position( parent, marker[ 1 ].start.path ),  new Position( parent, marker[ 1 ].end.path ) );
+				position.root.document.markers.set( marker[ 0 ], range );
+			}
+		// Otherwise we need to show warning about losing markers.
+		} else {
+			log.warn( 'model-writer-insert-lose-markers: Element containing markers is set to element without MarkersCollection.' );
+		}
+	}
```

Comment:
Will it work fine if I insert `DocumentFragment` with markers to another `DocumentFragment`?